### PR TITLE
Sync `v2-2-stable` with `v2-2-test` to release `2.2.2`

### PR DESCRIPTION
**Pull request by jedcunningham**
Time for `2.2.2rc1`!

### CODE REVIEW

File: CHANGELOG.txt [hunk 1]
```diff
@@ -1,3 +1,38 @@
+Airflow 2.2.1, 2021-11-11
```
Comment (potiuk) [inline CHANGELOG.txt]:
```suggestion
Airflow 2.2.2, 2021-11-11
```

Comment (jedcunningham) [reply CHANGELOG.txt]: 🤦‍♂️ thanks!

File: CHANGELOG.txt [hunk 2]
```diff
@@ -1,3 +1,38 @@
...
+- Add explicit session parameter in PoolSlotsAvailableDep (#18875)
+- FAB still requires WTForms < 3.0 (#19466)
+
+Doc only changes
+""""""""""""""""
+
+- Add missing parameter documentation for "timetable" (#19282)
+- Docs: Fix typo in ``dag-run.rst`` (#19340)
+- Improve Kubernetes Executor docs (#19339)
+- Fix typo in ``tutorial.rst`` (#18983)
```
Comment (kaxil) [inline CHANGELOG.txt]:
```suggestion
- Add missing parameter documentation for "timetable" (#19282)
- Improve Kubernetes Executor docs (#19339)
```

File: CHANGELOG.txt [hunk 3]
```diff
@@ -1,3 +1,38 @@
...
+- Bugfix: Check next run exists before reading data interval (#19307)
+- Fix MySQL db migration with default encoding/collation (#19268)
+- Fix hidden tooltip position (#19261)
+- sqlite_default has been hard-coded to /tmp, usegettempdir instead, (#19255)
+- Fix Toggle Wrap on DAG code page (#19211)
+- Clarify dag-not-found error message (#19338)
+- Add Note to SLA regarding schedule_interval (#19173)
+- Use ``execution_date`` to check for existing ``DagRun`` for ``TriggerDagRunOperator`` (#18968)
+- Add explicit session parameter in PoolSlotsAvailableDep (#18875)
+- FAB still requires WTForms < 3.0 (#19466)
```
Comment (kaxil) [inline CHANGELOG.txt]:
```suggestion
- FAB still requires ``WTForms<3.0`` (#19466)
```

File: CHANGELOG.txt [hunk 4]
```diff
@@ -1,3 +1,38 @@
...
+- Fix message on "Mark as" confirmation page (#19363)
+- Bugfix: Check next run exists before reading data interval (#19307)
+- Fix MySQL db migration with default encoding/collation (#19268)
+- Fix hidden tooltip position (#19261)
+- sqlite_default has been hard-coded to /tmp, usegettempdir instead, (#19255)
+- Fix Toggle Wrap on DAG code page (#19211)
+- Clarify dag-not-found error message (#19338)
+- Add Note to SLA regarding schedule_interval (#19173)
+- Use ``execution_date`` to check for existing ``DagRun`` for ``TriggerDagRunOperator`` (#18968)
+- Add explicit session parameter in PoolSlotsAvailableDep (#18875)
```
Comment (kaxil) [inline CHANGELOG.txt]:
```suggestion
- Add explicit session parameter in ``PoolSlotsAvailableDep`` (#18875)
```

File: CHANGELOG.txt [hunk 5]
```diff
@@ -1,3 +1,38 @@
...
+- Fix downgrade for a DB Migration (#19390)
+- Only mark SchedulerJobs as failed, not any jobs (#19375)
+- Fix message on "Mark as" confirmation page (#19363)
+- Bugfix: Check next run exists before reading data interval (#19307)
+- Fix MySQL db migration with default encoding/collation (#19268)
+- Fix hidden tooltip position (#19261)
+- sqlite_default has been hard-coded to /tmp, usegettempdir instead, (#19255)
+- Fix Toggle Wrap on DAG code page (#19211)
+- Clarify dag-not-found error message (#19338)
+- Add Note to SLA regarding schedule_interval (#19173)
```
Comment (kaxil) [inline CHANGELOG.txt]:
```suggestion
- Add Note to SLA regarding ``schedule_interval`` (#19173)
```

File: CHANGELOG.txt [hunk 6]
```diff
@@ -1,3 +1,38 @@
...
+- Task should fail immediately when pod is unprocessable (#19359)
+- Fix downgrade for a DB Migration (#19390)
+- Only mark SchedulerJobs as failed, not any jobs (#19375)
+- Fix message on "Mark as" confirmation page (#19363)
+- Bugfix: Check next run exists before reading data interval (#19307)
+- Fix MySQL db migration with default encoding/collation (#19268)
+- Fix hidden tooltip position (#19261)
+- sqlite_default has been hard-coded to /tmp, usegettempdir instead, (#19255)
+- Fix Toggle Wrap on DAG code page (#19211)
+- Clarify dag-not-found error message (#19338)
```
Comment (kaxil) [inline CHANGELOG.txt]:
```suggestion
- Clarify "dag not found" error message in CLI (#19338)
```

File: CHANGELOG.txt [hunk 7]
```diff
@@ -1,3 +1,38 @@
...
+- Fix serialization of Params with set data type (#19267)
+- Check if job object is None before calling .is_alive() (#19380)
+- Task should fail immediately when pod is unprocessable (#19359)
+- Fix downgrade for a DB Migration (#19390)
+- Only mark SchedulerJobs as failed, not any jobs (#19375)
+- Fix message on "Mark as" confirmation page (#19363)
+- Bugfix: Check next run exists before reading data interval (#19307)
+- Fix MySQL db migration with default encoding/collation (#19268)
+- Fix hidden tooltip position (#19261)
+- sqlite_default has been hard-coded to /tmp, usegettempdir instead, (#19255)
```
Comment (kaxil) [inline CHANGELOG.txt]:
```suggestion
- ``sqlite_default`` Connection has been hard-coded to `/tmp`, use `gettempdir` instead (#19255)
```

File: CHANGELOG.txt [hunk 8]
```diff
@@ -1,3 +1,38 @@
...
+
+Bug Fixes
+"""""""""
+
+- Fix bug when checking for existence of a Variable (#19395)
+- Fix Serialization when``relativedelta`` is passed as ``schedule_interval``  (#19418)
+- Fix moving of dangling TaskInstance rows for SQL Server (#19425)
+- Fix task instance modal in gantt view (#19258)
+- Fix serialization of Params with set data type (#19267)
+- Check if job object is None before calling .is_alive() (#19380)
```
Comment (kaxil) [inline CHANGELOG.txt]:
```suggestion
- Check if job object is ``None`` before calling ``.is_alive()`` (#19380)
```

File: CHANGELOG.txt [hunk 9]
```diff
@@ -1,3 +1,38 @@
...
+-------------------------
+
+Bug Fixes
+"""""""""
+
+- Fix bug when checking for existence of a Variable (#19395)
+- Fix Serialization when``relativedelta`` is passed as ``schedule_interval``  (#19418)
+- Fix moving of dangling TaskInstance rows for SQL Server (#19425)
+- Fix task instance modal in gantt view (#19258)
+- Fix serialization of Params with set data type (#19267)
```
Comment (kaxil) [inline CHANGELOG.txt]:
```suggestion
- Fix serialization of ``Params`` with set data type (#19267)
```

**Review by kaxil — APPROVED** [review]

**Comment by kaxil** [comment]
Marked it as a draft to avoid accidental "squash and merge"

File: CHANGELOG.txt (hunk 7 above)
Comment (jedcunningham) [inline CHANGELOG.txt]:
```suggestion
- ``sqlite_default`` Connection has been hard-coded to ``/tmp``, use ``gettempdir`` instead (#19255)
```

**Review by kaxil — APPROVED** [review]

**Pull request merged**